### PR TITLE
Uk/1446 inventory prices

### PR DESCRIPTION
**Pull request by stveep**
Fix for #1446 - this was due to the order not having the distributor set at the point where line items are added, meaning that the variant cannot be scoped to the distributor. I set it up to pass the distributor as an extra param, which can be used if the order distributor is not found.

Please go through the rest of the order process and check that the prices are ok - I think it should be ok once the line item has been added though.

### CODE REVIEW

File: app/controllers/spree/admin/line_items_controller_decorator.rb [hunk 1]
```diff
@@ -19,7 +19,12 @@ def index
 
   def create
     variant = Spree::Variant.find(params[:line_item][:variant_id])
-    OpenFoodNetwork::ScopeVariantToHub.new(@order.distributor).scope(variant)
+    dist = if @order.distributor
+      @order.distributor
+    elsif params[:distributor_id]
+      Enterprise.find(params[:distributor_id])
+    end
+    OpenFoodNetwork::ScopeVariantToHub.new(dist).scope(variant)
```
Comment (enricostano) [inline app/controllers/spree/admin/line_items_controller_decorator.rb]: Maybe you can move `dist` to a private method, something similar to:
```ruby
def distributor
  if @order.distributor
    @order.distributor
  elsif params[:distributor_id]
    Enterprise.find(params[:distributor_id])
  end
end
```
BTW, what happens when `dist` is `nil`?

**Comment by stveep** [comment]
Yes this was a pretty quick and dirty fix. I've put something together the way you suggested:
https://github.com/stveep/openfoodnetwork/tree/uk/1446-2-set-new-order-distributor

I'll probably have to do quite a bit of feature spec rewiring as we now have an extra refresh step, so let me know if this is what you had in mind.

**Comment by oeoeaio** [comment]
Yeah, something like that. Only suggestion would be that maybe we can simplify a bit by rendering a new page `set_distribution` from the controller when `new` is requested but `order.distribution_set?` is `false`. Then you can submit just the order_cycle and distributor and load the normal `new` view, without the need to have any logic about when to show/hide the distribution_fields partial in the view. This way you can remove an override, instead of adding one... Thoughts?

**Comment by daniellemoorhead** [comment]
Hey there, removing the Aus Review label from this as we're waiting on response to @oeoeaio's last comment.

**Comment by myriamboure** [comment]
Hi there! How is this moving forward? Just a question as we have some users here who are confused about the issue... but no pressure of course, that's great you are working on it @stveep thank you so much!

**Comment by stveep** [comment]
Hi @myriamboure - I am picking this up again and hopefully will finish at the weekend. I'm redoing it in the way rob suggested above.

**Comment by myriamboure** [comment]
Awesome, thank you @stveep :-)

**Comment by stveep** [comment]
Eventually superseded and fixed by #1575

**Pull request closed without merging**